docs/dev/table/common.md

```diff
@@ -768,11 +852,37 @@ A Table API or SQL query is translated when:
 
 Once translated, a Table API or SQL query is handled like a regular DataStream or DataSet program and is executed when `StreamExecutionEnvironment.execute()` or `ExecutionEnvironment.execute()` is called.
 
+</div>
+
+<div data-lang="Blink planner" markdown="1">
+Table API and SQL queries are translated into [DataStream]({{ site.baseurl }}/dev/datastream_api.html) programs whether their input is streaming or batch. A query is internally represented as a logical query plan and is translated in two phases: 
+
+1. Optimization of the logical plan, 
+2. Translation into a DataStream program.
+
+The behavior of translating  a query is different for `TableEnvironment` and `StreamTableEnvironment`.
+
+For `TableEnvironment`, A Table API or SQL query is translated when `TableEnvironment.execute()` is called, because `TableEnvironment` will optimize multiple-sinks into one DAG.
+
+while for `StreamTableEnvironment`, A Table API or SQL query is translated when:
+
+* a `Table` is emitted to a `TableSink`, i.e., when `Table.insertInto()` is called.
+* a SQL update query is specified, i.e., when `TableEnvironment.sqlUpdate()` is called.
+* a `Table` is converted into a `DataStream`.
+
+Once translated, a Table API or SQL query is handled like a regular DataStream program and is executed when `TableEnvironment.execute()` or `StreamExecutionEnvironment.execute()` is called.
+
+</div>
+</div>
+
 {% top %}
 
 Integration with DataStream and DataSet API
 -------------------------------------------
 
+Both Flink and Blink streaming jobs can integrate with the `DataStream` API. Only Flink batch job can integrate with the `DataSet API`, Blink batch job could not be combined with both.
```

Comment:
Please do not call them Flink and Blink jobs, they are all Flink jobs but with different planner (old and new planner).

Reply:
nit: legacy sounds better old

Reply:
I agree. However, we have used "old" in the API (`.useOldPlanner()`) and some other [documentation pages](https://ci.apache.org/projects/flink/flink-docs-master/dev/table/).